utils/utils.go

```diff
@@ -41,4 +43,61 @@ func GetAbsPath(path string) string{
 		}
 	}
 	return result
-}
\ No newline at end of file
+}
+
+func GetCurrentServerVersion() (string, string) {
+	var version, product string
+	rhndefault := "/etc/rhn/rhn.conf"
+	webpath := "/usr/share/rhn/config-defaults/rhn_web.conf"
+	altpath := "/usr/share/rhn/config-defaults/rhn.conf"
+
+	p, err := ScannerFunc(rhndefault, "product_name =")
+	if err != nil {
+		p, err = ScannerFunc(altpath, "product_name =")
+		if err != nil {
+			product = "SUSE Manager"
+		}
+	}
+	product = p
+	if product != "SUSE Manager" {
+		product = "uyuni"
+		v, err := ScannerFunc(rhndefault, "web.version.uyuni = ")
```

Comment:
this code block is repeated 3 times. Don't you create a helper method where of pass a list of files to look at, and the property name and iterate over that?